src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.spring.security;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username)
+            throws UsernameNotFoundException {
+        User user = userService.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException("Can't find user with this email: "
+                        + username));
+        UserBuilder userBuilder = withUsername(username);
+        userBuilder.password(user.getPassword());
+        userBuilder.authorities(user.getRoles().stream()
+                .map(r -> r.getRoleName().name())

```

Comment:
Did you test the application and it worker correctly with accesssing to endpoint for different users with different roles?
if not please consider to change this line:
```suggestion
                .map(r -> "ROLE_" + r.getRoleName().name())
```